src/supported.ts

```diff
@@ -3,12 +3,28 @@
 // ----------------------------------------------------------------------------
 
 import { Position, Range, TextDocument, workspace } from "vscode";
-import { configKeys, configPrefix, languageId } from "./constants";
-import { containsArmSchema } from "./schemas";
+import { armTemplateLanguageId, configKeys, configPrefix } from "./constants";
+import { containsArmSchema, containsParametersSchema } from "./schemas";
 
-export const armDeploymentDocumentSelector = [
-    { language: languageId, scheme: 'file' },
-    { language: languageId, scheme: 'untitled' } // unsaved files
+export const templateDocumentSelector = [
+    { language: armTemplateLanguageId, scheme: 'file' },
+    { language: armTemplateLanguageId, scheme: 'untitled' } // unsaved files
+];
+
+export const parameterDocumentSelector = [
+    { language: 'json', scheme: 'file' },
+    { language: 'json', scheme: 'untitled' },
+    { language: 'jsonc', scheme: 'file' },
+    { language: 'jsonc', scheme: 'untitled' },
+];
+
+export const templateOrParameterDocumentSelector = [
```

Comment:
Most of AzureTools.ts now registers intellisense providers that work for both JSON files (which may be parameter files) and arm-template files.